### ISSUE TITLE
`tab-size` broken in new code view

### ISSUE THREAD

**Issue by leranjun** [refined-github/refined-github#6461]
### Description

When the GitHub tab size preference is set to anything but 4, the tab display is broken under the new code view preview (the old one, without directory listing on the left side, is not affected).

<img width="1582" alt="Screenshot" src="https://user-images.githubusercontent.com/25704248/230397740-d3e0cec9-8604-4ecf-b8d3-36aed86e4350.png">

Upon inspection, it seems like GitHub added a read-only textarea element behind the existing code display, which is overriden by the catch-all rules in `tab-size`.

https://github.com/refined-github/refined-github/blob/c470d8103ebc9aa02735d4c0cb72d543333c5c06/source/features/tab-size.css#L3-L5

https://github.com/refined-github/refined-github/blob/c470d8103ebc9aa02735d4c0cb72d543333c5c06/source/features/tab-size.css#L11-L15

<img width="1593" alt="Screenshot 2023-04-06 at 9 51 13 PM" src="https://user-images.githubusercontent.com/25704248/230398646-8db6e316-d6d9-4ac6-8b14-5c772d00ea92.png">

### How to replicate the issue + URL

URL in the screenshot: https://github.com/refined-github/refined-github/blob/c470d8103ebc9aa02735d4c0cb72d543333c5c06/source/features/sync-pr-commit-title.tsx

"New Code Search and Code View" must be turned on under Feature preview.

### Extension version

23.3.31

### Browser(s) used

Chrome 111.0.5563.146 / macOS 13.3

**Comment by fregante** [comment]
This is now a native GitHub feature that you can customize in your profile. Our feature only fills some gaps

**Comment by leranjun** [comment]
> This is now a native GitHub feature that you can customize in your profile. Our feature only fills some gaps

I suppose this means I should disable the feature?

**Comment by fregante** [comment]
"Fills gaps" means that GitHub doesn't apply your setting in some cases so RGH does it for you. There's nothing to disable

**Comment by leranjun** [comment]
> "Fills gaps" means that GitHub doesn't apply your setting in some cases so RGH does it for you. There's nothing to disable

Yeah, I understand that, but the problem is that, after I've set my GitHub settings, RGH breaks in the code view (i.e. RGH overrides my GitHub settings). I cannot select code properly because the tab size is different between what is displayed and what I can select, if you refer to the screenshot above. I'm just not sure why this is being closed as not planned when it is clearly a bug with RGH…

**Comment by qubidt** [comment]
> "Fills gaps" means that GitHub doesn't apply your setting in some cases so RGH does it for you

In the case of the new code view, RGH is "filling the gaps" *incorrectly*. The new `textarea` element doesn't have a `data-tab-size` attribute (it sets `tab-size` in the element's `style` attribute). As @leranjun said, the `tab-size: var(--tab-size) !important` rule overrides the `textarea`'s tab-size style (it does not have a `data-tab-size` attribute), causing the mismatch between the two overlapping texts.

This issue should be reopened.

**Comment by qubidt** [comment]
This custom CSS seems to workaround the issue for now:

```css
textarea#read-only-cursor-text-area {
  --tab-size: revert;
}
```

**Comment by fregante** [comment]
Sorry I had misunderstood this report. This has now been hotfixed so you should get the fix the next time your browser fetches it (every 4 hours at most, needs a page reload afterwards)

**Comment by leranjun** [comment]
Looks like it's working for me!

**Comment by qubidt** [comment]
FYI, it looks like the hotfix is distributed, but the `--tab-size` is still incorrectly overridden for sizes not accounted for the in the [current version of `tab-size.css`](https://github.com/refined-github/refined-github/blob/6979e88e6a7b3273338a2651ac4b533609ef97df/source/features/tab-size.css)

![image](https://user-images.githubusercontent.com/6306455/230835026-4d72abd5-005f-4f36-a224-4b2d5e204d23.png)

I guess this will be fixed once #6480 has been merged and released
